### PR TITLE
EPMDJ-5822 Updated kotlin to 1.4.21

### PR DESCRIPTION
**Pull request by ViktoriiaLebedeva**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -57,14 +55,14 @@ kotlin {
             defaultSourceSet {
                 dependsOn(sourceSets.named("commonMain").get())
                 dependencies {
-                    implementation("org.jetbrains.kotlinx:kotlinx-serialization-protobuf-native:$serializationRuntimeVersion")
+                    implementation("org.jetbrains.kotlinx:kotlinx-serialization-protobuf:$serializationRuntimeVersion")
                     implementation("com.epam.drill:transport:$drillTransportLibVerison")
                     implementation("com.epam.drill.interceptor:http:$drillHttpInterceptorVersion")
-                    implementation("org.jetbrains.kotlinx:kotlinx-collections-immutable:0.3.2")
+                    implementation("org.jetbrains.kotlinx:kotlinx-collections-immutable:$kxCollectionImmutable")
```
Comment (viktorbll) [inline build.gradle.kts:61]: kxCollections

File: src/commonMain/kotlin/com/epam/drill/common/Serializer.kt [hunk 2]
```diff
@@ -5,9 +5,11 @@ import kotlinx.serialization.json.*
 import kotlin.native.concurrent.SharedImmutable
 
 @SharedImmutable
-val json = Json(JsonConfiguration.Stable)
+val json = Json {
+    encodeDefaults = true
+    allowStructuredMapKeys = true
```
Comment (viktorbll) [inline src/commonMain/kotlin/com/epam/drill/common/Serializer.kt:10]: Why this configs?

File: build.gradle.kts [hunk 3]
```diff
@@ -44,8 +42,8 @@ kotlin {
     sourceSets {
         commonMain {
             dependencies {
-                implementation("org.jetbrains.kotlin:kotlin-stdlib-common")
-                implementation("org.jetbrains.kotlinx:kotlinx-serialization-runtime-common:$serializationRuntimeVersion")
+                implementation("org.jetbrains.kotlinx:kotlinx-serialization-core:$serializationRuntimeVersion")
```
Comment (viktorbll) [inline build.gradle.kts:45]: This dependency is not required.

File: build.gradle.kts [hunk 4]
```diff
@@ -29,12 +30,9 @@ allprojects {
     tasks.withType<KotlinCompile> {
         kotlinOptions.allWarningsAsErrors = true
     }
-    tasks.withType<KotlinNativeCompile> {
-        kotlinOptions.allWarningsAsErrors = true
-    }
     configurations.all {
         resolutionStrategy.dependencySubstitution {
-            substitute(module("org.jetbrains.kotlinx:kotlinx-coroutines-core-native:$coroutinesVersion")).with(module("org.jetbrains.kotlinx:kotlinx-coroutines-core-native:$coroutinesVersion-native-mt"))
+            substitute(module("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutinesVersion")).with(module("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutinesVersion-native-mt"))
```
Comment (viktorbll) [inline build.gradle.kts:35]: Remove this.

**Review by viktorbll — CHANGES_REQUESTED** [review]

**Review by IgorKey — APPROVED** [review]

**Review by viktorbll — APPROVED** [review]

**Pull request merged**